lib/msal-common/src/app/module/AuthorizationCodeModule.ts

```diff
@@ -56,50 +64,47 @@ export class AuthorizationCodeModule extends AuthModule {
         }
 
         // Create and validate request parameters
-        const requestParameters = new ServerCodeRequestParameters(
-            acquireTokenAuthority,
-            this.clientConfig.auth.clientId,
-            request,
-            this.getRedirectUri(),
-            this.cryptoObj,
-            true
-        );
-
-        // Check for SSO
-        if (!requestParameters.isSSOParam(this.getAccount())) {
-            // TODO: Check for ADAL SSO
-        }
+        let requestParameters: ServerCodeRequestParameters;
+        try {
```

Comment:
is there any value in separating this into a couple of try/catches for each of these calls that should be wrapped in a try/catch?

Reply:
Yeah, I believe code that inside a `try` can't be optimized by the JS runtime, so `try` blocks should be limited to just parts that are expected to throw.

Reply:
I would be interested in which parts are expected to throw in this function.

Reply:
I think pretty much all of these operations have a possibility of throwing an error. What i want to prevent is any errors throwing and putting the app in a bad state, so i added a try catch around these ops. If there is a better way to do this, I am happy to hear it, but here are the following steps and possible errors thrown in the try/catch here:
1. creating code request parameters - throws validation errors
2. updating cache entries - throws errors if storage isn't implemented correctly.
3. populating query params - throws validation errors
4. Creation of the navigation url - may throw null pointers or authority discovery errors
5. token exchange parameter creation - may throw null pointers
6. base64 encoding errors - may throw framework issues

Reply:
If everything can realistically throw, then it's fine, just wanted to make sure that was the case